argusAI/event_similarity_processor.py

```diff
@@ -0,0 +1,274 @@
+from argus.backend.plugins.sct.testrun import SCTTestRun
+from argusAI.utils.event_message_sanitizer import MessageSanitizer
+from argusAI.utils.scylla_connection import ScyllaConnection
+
+SLEEP_INTERVAL = 300  # 5 minutes
+SIMILARITY_THRESHOLD = 0.90
+MAX_SIMILARS = 20
+DAYS_BACK = int(os.getenv('DAYS_BACK', 120))
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s [%(levelname)s] %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+
+EmbeddingEntry = namedtuple('EmbeddingEntry', ['run_id', 'event_index', 'embedding', 'start_time'])
+
+class BGE_SMALL(ONNXMiniLM_L6_V2):
+    """
+    Compact English text embedding model by BAAI, released Sep 2023.
+    Trained on large-scale paired data with RetroMAE and contrastive learning.
+    Produces high-quality 384-dim embeddings, optimized for efficiency in semantic search, classification, and clustering.
+    """
+    MODEL_NAME = "bge-small-en-v1.5"
+    DOWNLOAD_PATH = Path.home() / ".cache" / "chroma" / "onnx_models" / MODEL_NAME
+    EXTRACTED_FOLDER_NAME = "onnx"
+    ARCHIVE_FILENAME = "onnx.tar.gz"
+    MODEL_DOWNLOAD_URL = (
+        "https://scylla-qa-public.s3.us-east-1.amazonaws.com/ArgusAI/bge-small-en-v1.5/onnx.tar.gz"
+    )
+    _MODEL_SHA256 = "e7d1743b0c08f55c687cff6af696683398682f9ab4fb3cad1be644ee5553a72d"
+
+logger = logging.getLogger(__name__)
+stop_event = Event()
+
+class EventType:
+    ERROR = "ERROR"
+    CRITICAL = "CRITICAL"
+
+class EventProcessor:
+    def __init__(self):
+        """Initialize the EventProcessor with ChromaDB client and collections."""
+        self.chromadb_client = chromadb.Client(Settings(anonymized_telemetry=False))
+        self.collections = {
+            EventType.ERROR: self.chromadb_client.get_or_create_collection("test_error_events"),
+            EventType.CRITICAL: self.chromadb_client.get_or_create_collection("test_critical_events")
+        }
+        self.embedding_model = BGE_SMALL()
+        self.sanitizer = MessageSanitizer()
+        self.similars_found = 0
+        self.db = ScyllaConnection()
+        logger.info("Event Processor initialized with error and critical collections")
+
+    def find_similarities(self, event_type, embedding_entry, similarity_threshold, max_similars):
+        """
+        Find similar events in the collection and update the database.
+
+        Args:
+            event_type (str): Type of event (ERROR or CRITICAL)
+            embedding_entry (dict): Event data including run_id, event_index, and embedding
+            similarity_threshold (float): Minimum similarity score for matches
+            max_similars (int): Maximum number of similar events to find
+        """
+        collection = self.collections[event_type]
+        run_id = embedding_entry.run_id
+        event_index = embedding_entry.event_index
+        embedding = embedding_entry.embedding
+
+        logger.info(f"Finding similarities for run_id: {run_id}, event_index: {event_index}, type: {event_type}")
```

Comment:
f-string instead of `%s` formatter

Reply:
I think f-strings are just fine

and we want something like that enforce, it's with a linter (in pre-commit hooks)

otherwise it's just it's just a waste of energy/mental shares

Reply:
Okay